src/Controller/Component/RequestHandlerComponent.php

```diff
@@ -113,8 +96,20 @@ class RequestHandlerComponent extends Component
      */
     public function __construct(ComponentRegistry $registry, array $config = [])
     {
+        if (!isset($config['viewClassMap'])) {
+            $config['viewClassMap'] = [
+                'json' => 'Json',
+                'xml' => 'Xml',
+                'ajax' => 'Ajax'
+            ];
+        }
+        if (!isset($config['inputTypeMap'])) {
+            $config['inputTypeMap'] = [
+                'json' => ['json_decode', true],
+                'xml' => [[$this, 'convertXml']],
+            ];
+        }
```

Comment:
Would be a little cleaner as:

    $config += ['inputTypeMap' => [...]];

Worth deprecating [that function](https://github.com/cakephp/cakephp/blob/issue-7212/src/Controller/Component/RequestHandlerComponent.php#L223) and using an anonymous function? Or event making it static so that this config can also be in defaultConfig rather than hackily injected.